runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/stores/BlockStore.java

```diff
@@ -33,13 +34,14 @@
    * A stale data created by previous failed task should be handled during the creation of new block.
    *
    * @param blockId the ID of the block to create.
+   * @param memoryPoolAssigner  the MemoryPoolAssigner for memory allocation.
    * @return the created block.
    * @throws BlockWriteException for any error occurred while trying to create a block.
    *                             (This exception will be thrown to the scheduler
    *                             through {@link org.apache.nemo.runtime.executor.Executor} and
    *                             have to be handled by the scheduler with fault tolerance mechanism.)
    */
-  Block createBlock(String blockId) throws BlockWriteException;
+  Block createBlock(String blockId, MemoryPoolAssigner memoryPoolAssigner) throws BlockWriteException;
```

Comment:
Injecting `MemoryPoolAssigner` directly would be really nice! Just to inform you,  every `Store` including `LocalFileStore` and `GlusterFileStore` use `MemoryPoolAssigner` while creating `FileBlock`, since it uses `SerializedPartition` that is by default written in off-heap memory.